### ISSUE TITLE
Karma cannot start Chrome version 108.0.5359.125

### ISSUE THREAD

**Issue by thecheeto** [karma-runner/karma#3827]
After Chrome auto updated to version 108.0.5359.125, karma can no longer start chrome. It gives the below output and then tries to open chrome 2 more times. Each time it will open karma in a new chrome tab (not a window) but will never connect to it and won't actually run the tests. Both `ChromeHeadless` and `Firefox` seem to run fine which makes me think it has something to do with the latest Chrome version.

``` js
19 12 2022 11:20:58.275:INFO [karma-server]: Karma v6.3.20 server started at http://localhost:9876/
19 12 2022 11:20:58.275:INFO [launcher]: Launching browsers Chrome with concurrency unlimited
19 12 2022 11:20:58.285:INFO [launcher]: Starting browser Chrome
19 12 2022 11:20:58.287:DEBUG [launcher]: null -> BEING_CAPTURED
19 12 2022 11:20:58.289:DEBUG [temp-dir]: Creating temp dir at C:\Users\chrism4\AppData\Local\Temp\karma-77373876
19 12 2022 11:20:58.291:DEBUG [launcher]: C:\Program Files\Google\Chrome\Application\chrome.exe --user-data-dir=C:\Users\chrism4\AppData\Local\Temp\karma-77373876 --enable-automation --no-default-browser-check --no-first-run --disable-default-apps --disable-popup-blocking --disable-translate --disable-background-timer-throttling --disable-renderer-backgrounding --disable-device-discovery-notifications http://localhost:9876/?id=77373876
- Generating browser application bundles (phase: building)...19 12 2022 11:21:00.449:DEBUG [launcher]: Process Chrome exited with code 0 and signal null
19 12 2022 11:21:00.451:ERROR [launcher]: Cannot start Chrome

19 12 2022 11:21:00.455:ERROR [launcher]: Chrome stdout: Opening in existing browser session.

19 12 2022 11:21:00.456:ERROR [launcher]: Chrome stderr:
19 12 2022 11:21:00.456:DEBUG [temp-dir]: Cleaning temp dir C:\Users\chrism4\AppData\Local\Temp\karma-77373876
| Generating browser application bundles (phase: building)...19 12 2022 11:21:00.916:INFO [launcher]: Trying to start Chrome again (1/2).
19 12 2022 11:21:00.918:DEBUG [launcher]: BEING_CAPTURED -> RESTARTING
19 12 2022 11:21:00.921:DEBUG [launcher]: RESTARTING -> FINISHED
19 12 2022 11:21:00.922:DEBUG [launcher]: Restarting Chrome
19 12 2022 11:21:00.923:DEBUG [launcher]: FINISHED -> BEING_CAPTURED
19 12 2022 11:21:00.923:DEBUG [temp-dir]: Creating temp dir at C:\Users\chrism4\AppData\Local\Temp\karma-77373876
19 12 2022 11:21:00.925:DEBUG [launcher]: C:\Program Files\Google\Chrome\Application\chrome.exe --user-data-dir=C:\Users\chrism4\AppData\Local\Temp\karma-77373876 --enable-automation --no-default-browser-check --no-first-run --disable-default-apps --disable-popup-blocking --disable-translate --disable-background-timer-throttling --disable-renderer-backgrounding --disable-device-discovery-notifications http://localhost:9876/?id=77373876
| Generating browser application bundles (phase: building)...19 12 2022 11:21:02.515:DEBUG [launcher]: Process Chrome exited with code 0 and signal null
19 12 2022 11:21:02.523:ERROR [launcher]: Cannot start Chrome

19 12 2022 11:21:02.530:ERROR [launcher]: Chrome stdout: Opening in existing browser session.

19 12 2022 11:21:02.531:ERROR [launcher]: Chrome stderr:
19 12 2022 11:21:02.534:DEBUG [temp-dir]: Cleaning temp dir C:\Users\chrism4\AppData\Local\Temp\karma-77373876
- Generating browser application bundles (phase: building)...19 12 2022 11:21:03.422:INFO [launcher]: Trying to start Chrome again (2/2).
19 12 2022 11:21:03.422:DEBUG [launcher]: BEING_CAPTURED -> RESTARTING
19 12 2022 11:21:03.423:DEBUG [launcher]: RESTARTING -> FINISHED
19 12 2022 11:21:03.423:DEBUG [launcher]: Restarting Chrome
19 12 2022 11:21:03.423:DEBUG [launcher]: FINISHED -> BEING_CAPTURED
19 12 2022 11:21:03.424:DEBUG [temp-dir]: Creating temp dir at C:\Users\chrism4\AppData\Local\Temp\karma-77373876
19 12 2022 11:21:03.425:DEBUG [launcher]: C:\Program Files\Google\Chrome\Application\chrome.exe --user-data-dir=C:\Users\chrism4\AppData\Local\Temp\karma-77373876 --enable-automation --no-default-browser-check --no-first-run --disable-default-apps --disable-popup-blocking --disable-translate --disable-background-timer-throttling --disable-renderer-backgrounding --disable-device-discovery-notifications http://localhost:9876/?id=77373876
/ Generating browser application bundles (phase: building)...19 12 2022 11:21:05.869:DEBUG [launcher]: Process Chrome exited with code 0 and signal null
19 12 2022 11:21:05.874:ERROR [launcher]: Cannot start Chrome

19 12 2022 11:21:05.877:ERROR [launcher]: Chrome stdout: Opening in existing browser session.

19 12 2022 11:21:05.878:ERROR [launcher]: Chrome stderr:
19 12 2022 11:21:05.879:DEBUG [temp-dir]: Cleaning temp dir C:\Users\chrism4\AppData\Local\Temp\karma-77373876
- Generating browser application bundles (phase: building)...19 12 2022 11:21:08.628:ERROR [launcher]: Chrome failed 2 times (cannot start). Giving up.
19 12 2022 11:21:08.628:DEBUG [launcher]: BEING_CAPTURED -> FINISHED
```

### Environment Details
- karma versions:
    "karma": "~6.3.12",
    "karma-chrome-launcher": "~3.1.1",
    "karma-coverage": "~2.2.0",
    "karma-firefox-launcher": "^2.1.2",
    "karma-jasmine": "~4.0.0",
    "karma-jasmine-html-reporter": "^1.5.0",
    "karma-parallel": "^0.3.1",
    "karma-sabarivka-reporter": "^3.2.0",
    "karma-teamcity-reporter": "^1.1.0",
    "karma-verbose-reporter": "0.0.6"

- karma.conf.js:

``` js
// Karma configuration file, see link for more information
// https://karma-runner.github.io/1.0/config/configuration-file.html

module.exports = function (config) {
  config.set({
    basePath: '',
    frameworks: ['jasmine', '@angular-devkit/build-angular'],
    plugins: [
      require('karma-jasmine'),
      require('karma-chrome-launcher'),
      require('karma-firefox-launcher'),
      require('karma-jasmine-html-reporter'),
      require('karma-coverage'),
      require('karma-teamcity-reporter'),
      require('@angular-devkit/build-angular/plugins/karma')
    ],
    client: {
      clearContext: false, // leave Jasmine Spec Runner output visible in browser
      args: {
        random: false
      }
    },
    customLaunchers: {
      ChromeHeadlessNoSandbox: {
        base: 'ChromeHeadless',
        flags: [
          '--no-sandbox',
          '--disable-gpu',
          '--disable-web-security'
        ]
      }
    },
    jasmineHtmlReporter: {
      suppressAll: true // removes the duplicated traces
    },
    coverageReporter: {
      dir: require('path').join(__dirname, './coverage/ngfnex-ui'),
      subdir: '.',
      fixWebpackSourcePaths: true,
      reporters: [
        { type: 'html' },
        { type: 'text-summary' }
      ],
      check: {
        emitWarning: true,
        global: {
          statements: 80,
          branches: 80,
          functions: 80,
          lines: 80
        }
      }
    },
    reporters: ['progress', 'kjhtml', 'teamcity'],
    port: 9876,
    colors: true,
    logLevel: config.LOG_DEBUG,
    autoWatch: true,
    browsers: ['Chrome'],
    singleRun: false,
    restartOnFileChange: true
  });
};
```

### Steps to reproduce the behaviour

1. Update to chrome to latest version (108.0.5359.125)
2. Run `ng test` in angular project using karma

**Comment by PigeonPanet** [comment]
I also have this issue

**Comment by thecheeto** [comment]
@PigeonPanet have you been able to find a fix for this? I'm still seeing this issue in Chrome version 111.0.5563.111.

**Comment by demike** [comment]
Same issue here on Chrome 111.0.5563.65

**Comment by thecheeto** [comment]
I finally figured this issue out! Somehow chrome got this policy https://chromeenterprise.google/policies/?policy=UserDataDir added that will ignore the `--user-data-dir` flag and use whatever is set in there. Not sure how it got there but might of been added through my organization. This caused the user data dir that karma was generating to be completely ignored and therefore not work.

To fix, I deleted the registry entry from here:
<img width="855" alt="image" src="https://github.com/karma-runner/karma/assets/2992939/853f4ef2-5c2f-4e96-9ca9-cebbba10471d">

I did uninstall Chrome first and then reinstalled it after I deleted it, but I'm not sure if that's required.
